_posts/2020-02-11-spring-boot-openapi.md

```diff
@@ -0,0 +1,394 @@
+
+### Tags
+
+Then comes some additional metadata about our API.
+
+```yaml
+tags:
+- name: user
+  description: Operations about user
+  externalDocs:
+    description: Find out more about our store
+    url: http://swagger.io
+```
+
+* The [`tags`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#fixed-fields) section, provide additional metadata, which can be used to make our API more readable and easier to follow. Each tag should be unique.
+
+### Paths
+
+Here we will describe some paths, to make it clean how we can create our own. But, what are paths? Relative paths holds the information  to the individual endpoints and their operations.
+
+```yaml
+paths:
+  /user:
+    post:
+      tags:
+      - user
+      summary: Create user
+      description: Create user functionality
+      operationId: createUser
+      requestBody:
+        description: Created user object
+        content:
+          '*/*':
+            schema:
+              $ref: '#/components/schemas/User'
+        required: true
+      responses:
+        default:
+          description: successful operation
+          content: {}
+      x-codegen-request-body-name: body
+  /user/{username}:
+    get:
+      tags:
+      - user
+      summary: Get user by user name
+      operationId: getUserByName
+      parameters:
+      - name: username
+        in: path
+        description: 'The name that needs to be fetched. Use user1 for testing. '
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+        404:
+          description: User not found
+          content: {}
+```
+
+* First the [`ref`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#fixed-fields-7) field, allows us to refer own defined Schema. In this case the User schema.
+* Then the `summary`, which is optional and is intended to apply to all operations in this path.
+* Next is the [`operationId`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#fixed-fields-8), which is a unique string and is used to identify the operation. Think about it as our method name.
+* Last the [`responses`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#responsesObject) object.It must define at least one successful response code for an operation call. 
+
+### Components
+
+The objects of the API are all described here. The objects defined within the components object will have no effect on the API unless they are explicitly referenced from properties outside the components object.
+
+```yaml
+components:
+  schemas:
+    User:
+      type: object
+      properties:
+        id:
+          type: integer
+          format: int64
+        username:
+          type: string
+        firstName:
+          type: string
```

Comment:
I removed some of those string attributes as this seems quite repetitive for the reader.